### PR TITLE
Switch kernel for lp

### PR DESCRIPTION
**Pull request by panlinux**
This closes issue #146 by adding an option (not default) to allow the script to install a new kernel if the currently running one is deemed incompatible with livepatch.

https://launchpad.net/~ahasenack/+archive/ubuntu/ua-livepatch-kernel-switch-pr147 is building packages if you want to try it out. Note that livepatch is only available for LTS releases.

### CODE REVIEW

**Comment by panlinux** [comment]
Before merging I'll cleanup the commit history.

File: modules/service-livepatch.sh [hunk 1]
```diff
@@ -3,25 +3,92 @@
...
+        exit ${result}
+    fi
+}
+
+_old_kernel_message() {
+    local token="$1"
+
+    echo
+    echo "Your currently running kernel ($KERNEL_VERSION) is too old to"
+    echo "support snaps. Version 4.4.0 or higher is needed."
```
Comment (basak) [inline modules/service-livepatch.sh]: The user asked for livepatch, and doesn't know what snaps are or what they have to do with livepatch. What's this message about snaps? :)

Comment (panlinux) [reply modules/service-livepatch.sh]: I could clarify that Livepatch is delivered as a snap, and snaps require a newer kernel. Note this is not a message introduced in this branch, I just moved it to its own function to make the code easier to read.

I grant that there might be some confusion between this kernel update for snaps in general, and the other kernel *change* for livepatch support. Unfortunately I cannot determine if a kernel is suitable for livepatching before I install the canonical-livepatch snap. That snap is what tells me if the running kernel can be livepatched or not.

**Comment by panlinux** [comment]
Thanks for the review so far.
Correct, making sure the new kernel is what we are booting into will be done in another PR.

File: ubuntu-advantage.1 [hunk 2]
```diff
@@ -79,9 +79,16 @@ Managed live kernel patching. For more information, visit
 https://www.ubuntu.com/server/livepatch
 .TP
 .B
-enable-livepatch \fI<token>\fR
+enable-livepatch \fI<token>\fR [\fB\-\-allow\-kernel\-change\fR]
```
[the remark this answers is not in the document]
Comment (panlinux) [reply ubuntu-advantage.1]: Yes it does? It's an optional parameter, like in the other cases:
`disable-livepatch \fR[\fB\-r\fR]`
`enable-fips-updates \fItoken\fR [\fB\-y\fR]`

[the remark this answers is not in the document]
Comment (panlinux) [reply ubuntu-advantage.1]: Ah, the token. n/m, fixing.

**Pull request merged**